clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -680,6 +790,7 @@ private long maybeUpdate(long now, Node node) {
         final RequestCompletionHandler callback;
         final boolean expectResponse;
         final boolean isInternalMetadataRequest; // used to flag metadata fetches which are triggered internally by NetworkClient
+        final boolean isInternalApiVersionRequest; // used to flag api version fetches which are triggered internally by NetworkClient
```

Comment:
One can argue against it, as now the same check has to be done at multiple places. If number of such internal requests grow beyond two, then probably it won't make sense to keep a flag for each of them. At that point it would be better to add a method like `boolean isInternalRequestOfType(ApiKeys apiKey)`. Makes sense?